logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt

```diff
@@ -47,5 +50,53 @@ class ServerPassword (
 
     val usernameField: String? = null,
     val passwordField: String? = null
-)
+) {
 
+
+    companion object {
+        fun fromJSON(jsonObject: JSONObject): ServerPassword {
+
+            fun stringOrNull(s: String): String? {
+                try {
+                    return jsonObject.getString(s)
+                } catch(e: JSONException) {
+                    return null
+                }
+            }
+
+            return ServerPassword(
+                    id = jsonObject.getString("id"),
+
+                    hostname = jsonObject.getString("hostname"),
+                    password = jsonObject.getString("password"),
+                    username = stringOrNull("username"),
+
+                    httpRealm = stringOrNull("httpRealm"),
+                    formSubmitURL = stringOrNull("formSubmitURL"),
+
+                    usernameField = stringOrNull("usernameField"),
+                    passwordField = stringOrNull("passwordField"),
+
+                    timesUsed = jsonObject.getInt("timesUsed"),
+
+                    timeCreated = jsonObject.getLong("timeCreated"),
+                    timeLastUsed = jsonObject.getLong("timeLastUsed"),
+                    timePasswordChanged = jsonObject.getLong("timePasswordChanged")
+            )
+        }
+
+        fun fromJSON(jsonText: String): ServerPassword {
```

Comment:
I don't know a lot about Kotlin but it's weird to me that this doesn't include exception information.

Reply:
Yep, Kotlin doesn't check exceptions.